### PR TITLE
Add testfreeze plugin

### PR DESCRIPTION
**Pull request by saschagrunert**
This adds a new plugin to add conditional information during test freeze period. This is part of https://github.com/kubernetes/release/issues/2386, while we will request cherry-picks as intermediate step until the auto fast-forward goes live (planned for the v1.25 cycle)

Supersedes https://github.com/kubernetes/test-infra/pull/25300
cc @kubernetes/release-engineering

### CODE REVIEW

**Comment by saschagrunert** [comment]
@petr-muller @alvaroaleman PTAL when the CI is happy

**Comment by saschagrunert** [comment]
I think we need a new checkconfig image for this.

File: config/prow/plugins.yaml [hunk 1]
```diff
@@ -831,6 +831,7 @@ plugins:
     - size
     - skip
     - slackevents
+    - testfreeze
```
Comment (chaodaiG) [inline config/prow/plugins.yaml:834]: this is a chicken and egg problem. note that the checkconfig job uses a static image so it won't understand testfreeze plugin until this PR is merged (which would trigger a rebuild of checkconfig image, then there will be an automated process to update checkconfig job)

File: prow/plugins/testfreeze/testfreeze_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+func TestHandle(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range []struct {
+		action            github.PullRequestEventAction
+		org, repo, branch string
+		prepare           func(*testfreezefakes.FakeImpl)
+		assert            func(*testfreezefakes.FakeImpl, error)
+	}{
+		{ // success in test freeze
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply prow/plugins/testfreeze/testfreeze_test.go]: Yep, improved as suggested :+1:

File: prow/plugins/testfreeze/testfreeze.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate . impl
+type impl interface {
+	CheckInTestFreeze(*logrus.Entry) (*checker.Result, error)
+	CreateComment(plugins.PluginGitHubClient, string, string, int, string) error
+}
+
+type defaultImpl struct{}
```
Comment (stevekuznetsov) [inline prow/plugins/testfreeze/testfreeze.go:96]: nit: `impl` and `defaultImpl` are not informative names. Why not

```go
type Checker struct {
  checker
}

type checker interface{/*...*/}

type defaultChecker struct {/*...*/}
```
?

Comment (saschagrunert) [reply prow/plugins/testfreeze/testfreeze.go]: Yes, changed that for the `checker` package. For the `testfreeze` package I had to choose a different name (`verifier`) to not clash with the `checker` package.

**Comment by saschagrunert** [comment]
@chaodaiG @stevekuznetsov @alvaroaleman please take a look again.

**Review by chaodaiG — APPROVED** [review]

**Pull request merged**